### PR TITLE
[CARBONDATA-654] Add data update and deletion example

### PR DESCRIPTION
**Pull request by chenliang613**
Add data update and deletion example

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/623/

File: examples/spark/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+           """)
+
+    // Query data before update and deletion
+    cc.sql("""
+           SELECT * FROM t3
+           """).show()
+
+    // Delete data where salary > 15005
+    cc.sql("""
+           delete FROM t3 where salary > 15005
```
Comment (jackylk) [inline examples/spark/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: Please use capital letters

Comment (jackylk) [inline examples/spark/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: Please use capital letters

Comment (jackylk) [inline examples/spark/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: Please use capital letters

File: examples/spark/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+           """).show()
+
+    // Query data again after delete data where salary > 15005
+    cc.sql("""
+           SELECT * FROM t3
+           """).show()
+
+    // Update data where salary < 15003
+    cc.sql("""
+           update t3 set (t3.country) = ('india') where t3.salary < 15003
```
Comment (jackylk) [inline examples/spark/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: can you add a more complex update statement, like using JOIN and subquery?

Comment (chenliang613) [inline examples/spark/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: ok

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/654/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/658/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/662/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/665/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**